src/app/technical-record-create/technical-record-create.component.ts

```diff
@@ -20,41 +19,54 @@ export class TechnicalRecordCreateComponent implements OnInit {
   vehicleTypes = ['PSV', 'HGV', 'Trailer'];
   createTechRecordForm: FormGroup;
   formError$: Observable<any>;
-  formErrors: { vinErr: string, vrmErr: string, vTypeErr: string, requestErr: string[] } = {
+  formErrors: { vinErr: string; vrmErr: string; vTypeErr: string; requestErr: string[] } = {
     vinErr: '',
     vrmErr: '',
     vTypeErr: '',
-    requestErr: ['']
+    requestErr: []
   };
   hasErrors = false;
   hasVinReqErrors;
   hasVrmReqErrors;
   vrmLabel = CREATE_PAGE_LABELS.CREATE_VRM_LABEL;
 
   constructor(private _store: Store<IAppState>) {
-    this.formError$ = this._store.select(s => s.vehicleTechRecordModel.initialDetails.error);
+    this.formError$ = this._store.select(getErrors);
   }
 
   ngOnInit() {
     this.createTechRecordForm = new FormGroup({
-      'vehicleType': new FormControl('', Validators.required),
-      'vin': new FormControl('', [Validators.required, Validators.minLength(1), Validators.maxLength(21)]),
-      'vrm': new FormControl('', Validators.required)
+      vehicleType: new FormControl('', Validators.required),
+      vin: new FormControl('', [
+        Validators.required,
+        Validators.minLength(1),
+        Validators.maxLength(21)
+      ]),
+      vrm: new FormControl('', Validators.required)
     });
 
-    this.formError$.subscribe(result => {
-      this.formErrors.requestErr = result;
-      this.hasVinReqErrors = result.includes(CREATE_PAGE_LABELS.CREATE_VIN_LABEL_ERROR);
-      this.hasVrmReqErrors = result.includes(CREATE_PAGE_LABELS.CREATE_VRM_LABEL_ERROR);
+    this.formError$.subscribe((result) => {
```

Comment:
if you are subscribing manually you have to clean up otherwise you will be causing memory leak. Better if you pass `formError$ | async` then you dont have to unsubscribe. Code will feel cleaner too.